# Hypothesis testing problems

## Exercise 1

You are a nutritionist researching two different types of diets to see if there is a significant difference in weight loss after one month. You choose two random groups of people; one group follows the first diet, and the other follows the second. At the end of the month, you record the weight loss (in kg) for each person:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

With these data, it seeks to answer the following question: Is there a significant difference in average weight loss between people who followed the first diet and those who followed the second diet?

To draw conclusions, follow the points below:

- State the hypothesis: null and alternative hypothesis.
- Perform the test to test the hypothesis. You can use a Student's t-test.
- Analyze the conclusions.

### State the hypotheses
- **H0:** There is not difference in average weight loss between the two diets
- **H1:** There is a difference in average weight loss betweeen the two diets

In [1]:
# TODO
import numpy as np
import scipy.stats as stats

diet_1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
diet_2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]

# The data is less than 30, so the Student's test must be applied
t_value, p_value, = stats.ttest_ind(diet_1, diet_2)

print(f"t-value: {t_value}")
print(f"p_value: {p_value}")

t-value: -3.5383407969933938
p_value: 0.004083270191713912


### Conclusion
- **t-value:** In this case, the t-value is negative, indicating that the population mean is significantly less than the sample mean. The absolute value of the t-value (3.538) indicates the magnitude of the difference between the population mean and the sample mean.
- **p_value:** Is very low (0.004083), indicating that the probability that the observed difference is due to chance is very small. Typically, if the p-value is less than a given significance level (e.g., 0.05), the null hypothesis is rejected.

It is concluded that the population mean is significantly different from the sample mean.

## ANOVA

**ANOVA** (*Analysis of Variance*) is a statistical technique used to compare the measures of two or more groups. The idea behind ANOVA is to decompose the total variability in the data into two components: between-group variability and within-group variability:

- **Between-group variability**: This variability refers to the differences between the group means. If this variability is considerably larger than the within-group variability, it could be an indication that at least one of the group means is different.
- **Within-group variability**: This variability refers to the dispersion of the data within each group. If all groups have similar variability, then any noticeable difference in group means could be considered significant.

Hypotheses in ANOVA typically include:

- **Null hypothesis** ($H₀$): The means of all groups are equal.
- **Alternative hypothesis** ($H₁$): At least one of the group means is different.

If the ANOVA test result is significant (e.g., a p-value less than a threshold such as 0.05), this suggests that at least one group mean is different.

## Exercise 2

A farmer decides to test three different types of fertilizers to determine if one is superior in terms of corn production. The farmer plants corn on 15 identical plots and uses all three fertilizers (5 plots for each type). At the end of the season, he measures the corn yield (in kg) of each plot, with the following result:

| Fertilizer 1 | Fertilizer 2 | Fertilizer 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

With this data, he seeks to answer the following question: Is there a significant difference in average corn yield between the three types of fertilizers?

To help you, follow the points below:

- State the hypothesis: null and alternative hypothesis.
- Perform the ANOVA test.
- Analyze the conclusions.
- If one fertilizer is better than another, how can we know it?

### State the hypotheses
- **H0:** The corn yield averages of the three fertilizers are equal.
- **H1:** At least one of the fertilizer corn yield averages is different.

In [3]:
# TODO
import scipy.stats as stats

f_1 = [20, 21, 20, 19, 20]
f_2 = [22, 21, 23, 22, 21]
f_3 = [24, 23, 22, 23, 24]

# There are three samples to analyze so we will use ANOVA test
f_value, p_value_2, = stats.f_oneway(f_1, f_2, f_3)

print(f"f_value: {f_value}")
print(f"p_value: {p_value_2}")

f_value: 20.315789473684188
p_value: 0.00014047824793190475


### Conclusion

The results suggest that the relationship between the variables is significant and that the probability that this relationship is due to chance is extremely low.
There is a significant difference in mean corn yield between at least two of the fertilizers.

### Determining the best fertilizer

We will use the Tukey test to determine which is the best fertilizer

In [4]:
import numpy as nop
from statsmodels.stats.multicomp import pairwise_tukeyhsd

data = np.concatenate([f_1, f_2, f_3])
labels = ["f1"] * 5 + ["f2"] * 5 + ["f3"] * 5

# Applyig Tukey test
result = pairwise_tukeyhsd(data, labels, alpha= 0.05)
print(result)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower  upper  reject
--------------------------------------------------
    f1     f2      1.8 0.0099 0.4572 3.1428   True
    f1     f3      3.2 0.0001 1.8572 4.5428   True
    f2     f3      1.4 0.0409 0.0572 2.7428   True
--------------------------------------------------


### Conclusion

The reject column indicates that there is a significant difference between the three fertilizers. Therefore, we must check the difference between the means of each group.
It is observed that the mean of F2 differs and is greater than that of F1 and at the same time the mean of F3 is greater than that of F2. So we conclude that the best fertilizer is F3.